### DISCUSSION TITLE
Best Way to Integrate Python with Gama

### DISCUSSION THREAD

**Discussion by hana-gharrad** [gama-platform/gama#60] [Q&A]
Hello again,

Is there a plugin that could be used to integrate (call) python from Gama platform. I checked the Gampy plugin presented in Gama days 2022. My code in python is not a TensorFlow or Keras trained model, I can still use Gampy? Otherwise what is the best way to call a python or Matlab code and collect the generated results to feed it to a Gama agent?

Thank you so much.

**Comment by RoiArthurB** [comment]
Hi @hana-gharrad 

I guess the package Gampy is your way to go. From what I remember, the use of TensorFlow was a proof of concept over some pretty complex and heavy long process, but it should work fine for just calling a few python functions. For Mathlab, you have a plugin in `gama-platform/gama.experimental` for this.

Hope it'll all work well for you :)

**Reply by hana-gharrad** [reply]
Dear @RoiArthurB,

Thank you for your reply. Should I use the Gama Git version or the release version to work with Gampy. I could give a python file instead of the model file (will that work)?

Many thanks,
Kind regards,

**Reply by RoiArthurB** [reply]
I really can't tell you as I never used Gampy 😅😅😅 

You'll probably have better help by contacting Gampy mainteners :)

**Reply by hana-gharrad** [reply]
Thank you @RoiArthurB, I will try to investigate other options.

**Comment by benoitgaudou** [comment]
Dear,
Considering GamPy, it was dedicated to execute ml model on GAMA, and not to
run any generic python code.

The main issue with gampy was the size of the plug-in ( in particular due
to external jar files).

I am trying to have a plug-in based on maven dependency to avoid this issue.

By the way , right now the way to use python code with GAMA model is to
execute a python script (with the exec gaml operator), make your script
write a file and read the file from gama…
Another way could be to have a web server with your python code (eg a flask
server is really simple du implement) and have exchange  through http
request …

Cheers

Benoit

Le ven. 14 avr. 2023 à 17:33, Arthur Brugière ***@***.***> a
écrit :

**Reply by hana-gharrad** [reply]
Dear Benoit,

Thank you for the suggestion. I see flask with http is a nice option. I will give it it try. 

Kind regards,

**Comment by benoitgaudou** [comment]
Hi,
I just noticed that there is an example of Flask Python launcher in the
models of Network (in the script).
you will have to call your Python functions from the function associated to
a route, and perhaps have to transform your results in string (that can be
a json file by the way) before sending them back to GAMA.
Cheers

Benoit

Le dim. 16 avr. 2023 à 07:49, hana-gharrad ***@***.***> a
écrit :
